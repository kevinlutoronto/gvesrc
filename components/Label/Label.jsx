// @flow
import React from 'react';
import cx from 'classnames';

import { getDataAttrs } from '../../dataUtils';
import type { Data, GlobalAttributes } from '../../types';

import styles from './style.scss';

type Props = {|
  children?: React$Node,
  data?: Data,
  disabled?: boolean,
  htmlFor?: string,
  inline?: boolean,
  required?: boolean,
  text?: string,
  ...GlobalAttributes,
|};

function Label({
  children,
  className,
  data,
  disabled,
  htmlFor,
  inline,
  required,
  text,
  ...rest
}: Props) {
  const classes = cx(styles.label, className, {
    [styles.labelInline]: inline,
    [styles.labelDisabled]: disabled,
    [styles.labelRequired]: required,
  });

  return (
    <label className={classes} htmlFor={htmlFor} {...getDataAttrs(data)} {...rest}>
      {children || text}
    </label>
  );
}

Label.displayName = 'Plasma@Label';

export default Label;
