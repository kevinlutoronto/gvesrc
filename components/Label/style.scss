@import '../../styles/base';

.label {
  @extend %type--label;

  display: block;
  font-weight: bold;
  line-height: 20px;

  &[for] {
    cursor: pointer;
  }

  &:not(.labelInline) + * {
    margin-top: 5px;
  }

  &.labelDisabled {
    color: $cGray300;
    cursor: not-allowed;
  }

  &.labelInline {
    display: inline-block;
    margin-right: 20px;
    padding: 10px 0;
  }

  &.labelRequired::after {
    content: ' *';
  }
}
