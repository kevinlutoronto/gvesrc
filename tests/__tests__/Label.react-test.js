/* eslint-disable */

import React from 'react';
import renderer from 'react-test-renderer';

import Label from '../../components/Label/Label';

describe('Label Component', () => {
  test('render', () => {
    const component = renderer.create(<Label>Test</Label>);

    expect(component).toMatchSnapshot();
  });

  test('render inline', () => {
    const component = renderer.create(<Label inline>Test</Label>);

    expect(component).toMatchSnapshot();
  });

  test('render disabled', () => {
    const component = renderer.create(<Label disabled>Test</Label>);

    expect(component).toMatchSnapshot();
  });

  test('render required', () => {
    const component = renderer.create(<Label required>Test</Label>);

    expect(component).toMatchSnapshot();
  });
});
