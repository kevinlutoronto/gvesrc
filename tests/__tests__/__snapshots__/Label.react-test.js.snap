// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component render 1`] = `
<label
  className="label"
  htmlFor={undefined}
>
  Test
</label>
`;

exports[`Label Component render disabled 1`] = `
<label
  className="label labelDisabled"
  htmlFor={undefined}
>
  Test
</label>
`;

exports[`Label Component render inline 1`] = `
<label
  className="label labelInline"
  htmlFor={undefined}
>
  Test
</label>
`;

exports[`Label Component render required 1`] = `
<label
  className="label labelRequired"
  htmlFor={undefined}
>
  Test
</label>
`;
